src/tree.py

```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
+from __future__ import annotations
+
+
+@dataclass
+class Knæ:
+    parent: Knæ | None
+    ben: tuple[int, int] | None
+    children: dict[str, Knæ] | int
+
+
+class SuffixTree:
+    root: Knæ | None
+
+    def construct(self, x: str) -> None:
```

Comment:
You probably want this to be a static or class method. You need an instance to construct a suffix tree otherwise. If you first have to create the suffix tree and then call `construct()` on it, then the name is a bit misleading.